File: app/src/main/java/br/com/jonathanzanella/myexpenses/account/transactions/TransactionsView.java
package br.com.jonathanzanella.myexpenses.account.transactions;

import android.content.Context;
import android.util.AttributeSet;

import org.joda.time.DateTime;

import br.com.jonathanzanella.myexpenses.R;
import br.com.jonathanzanella.myexpenses.account.Account;
import br.com.jonathanzanella.myexpenses.views.BaseView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class TransactionsView extends BaseView {
	@BindView(R.id.view_account_transactions_this_month)
	MonthTransactionsView thisMonth;
	@BindView(R.id.view_account_transactions_next_month)
	MonthTransactionsView nextMonth;

	public TransactionsView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public TransactionsView(Context context, AttributeSet attrs, int defStyleAttr) {
		super(context, attrs, defStyleAttr);
	}

	protected void init() {
		inflate(getContext(), R.layout.view_account_transactions, this);
		ButterKnife.bind(this);
	}

	public void showTransactions(final Account account, final DateTime monthToShow) {
		thisMonth.setLoadTransactionsCallback(new LoadTransactionsCallback() {
			@Override
			public void onTransactionsLoaded(int balance) {
				nextMonth.showBalance(account, monthToShow.plusMonths(1), balance);
			}
		});

		thisMonth.showBalance(account, monthToShow, account.getBalance());
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/account/transactions/TransactionsView.kt
package br.com.jonathanzanella.myexpenses.account.transactions

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import br.com.jonathanzanella.myexpenses.R
import br.com.jonathanzanella.myexpenses.account.Account
import kotlinx.android.synthetic.main.view_account_transactions.view.*
import org.joda.time.DateTime

class TransactionsView: FrameLayout {

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    init {
        View.inflate(context, R.layout.view_account_transactions, this)
    }

    fun showTransactions(account: Account, monthToShow: DateTime) {
        thisMonth.setLoadTransactionsCallback(object: LoadTransactionsCallback {
            override fun onTransactionsLoaded(balance: Int) {
                nextMonth.showBalance(account, monthToShow.plusMonths(1), balance)
            }
        })

        thisMonth.showBalance(account, monthToShow, account.balance)
    }
}